### PR TITLE
Threat Hunting Playbooks - FireEye ETP and FireEye HX

### PR DESCRIPTION
**Pull request by TalNos**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-5011

## Description
New threat hunting playbooks for FireEye ETP and FireEye HX.

## Screenshots
'FireEye HX - Indicators Hunting' playbook:
FireEye_HX_-_Indicators_Hunting

'FireEye HX - Traffic Indicators Hunting' playbook:
FireEye_HX_-_Traffic_Indicators_Hunting

'FireEye HX - File Indicators Hunting' playbook:
FireEye_HX_-_File_Indicators_Hunting

'FireEye HX - Execution Flow Indicators Hunting' playbook:
FireEye_HX_-_Execution_Flow_Indicators_Hunting

'FireEye ETP - Indicators Hunting' playbooks:
FireEye_ETP_-_Indicators_Hunting

### CODE REVIEW

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 1]
```diff
@@ -0,0 +1,1417 @@
...
+- key: limit
+  value: {}
+  required: false
+  description: Limits the results count. (Once the limit is reached, the search is stopped).
+  playbookInputQuery:
+outputs:
+- contextPath: FireEyeHX.HuntingResults
+  description: Email message objects and fields that were retrieved from FireEye HX.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Timestamp - Event
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1361]: Is this contextPath correct?  Is there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 2]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The timestamp of the event.
+  type: number
+- contextPath: FireEyeHX.HuntingResults.Timestamp - Accessed
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1364]: Is this contextPath correct?  Is there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 3]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The last accessed time.
+  type: number
+- contextPath: FireEyeHX.HuntingResults.Timestamp - Modified
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1367]: Is this contextPath correct?  Is there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 4]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The time when the entry was last modified.
+  type: number
+- contextPath: FireEyeHX.HuntingResults.File Name
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1370]: Is this contextPath correct?  Is there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 5]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The name of the file.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.File Full Path
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1373]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 6]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The full path of the file.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.DNS Hostname
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1376]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 7]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The name of the DNS host.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.URL
+  description: The event URL.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Username
+  description: The event username.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.File MD5 Hash
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1385]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 8]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The MD5 hash of the file.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Port
+  description: The port.
+  type: number
+- contextPath: FireEyeHX.HuntingResults.Process ID
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1391]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 9]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The ID of the process.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Local IP Address
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1394]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 10]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The local IP address.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Local Port
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1397]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 11]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The local port.
+  type: number
+- contextPath: FireEyeHX.HuntingResults.Remote Port
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1400]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 12]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The remote port.
+  type: number
+- contextPath: FireEyeHX.HuntingResults.IP Address
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1403]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml [hunk 13]
```diff
@@ -0,0 +1,1417 @@
...
+  description: The IP address.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Process Name
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting.yml:1406]: Is this contextPath correct?  Are there really spaces in the contextPath?

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting_README.md [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+| interval_in_seconds | The interval in seconds between each poll. Default is 60.	 |  | Optional |
+| limit | Limits the results count. \(Once the limit is reached, the search is stopped\). |  | Optional |
+
+## Playbook Outputs
+---
+
+| **Path** | **Description** | **Type** |
+| --- | --- | --- |
+| FireEyeHX.HuntingResults | Email message objects and fields that were retrieved from FireEye HX. | string |
+| FireEyeHX.HuntingResults.Timestamp - Event | The timestamp of the event. | number |
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Execution_Flow_Indicators_Hunting_README.md:41]: Should there be spaces in the ContextPath?  Please check all the ContextPaths.

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_File_Indicators_Hunting.yml [hunk 15]
```diff
@@ -0,0 +1,2017 @@
...
+- key: limit
+  value: {}
+  required: false
+  description: Limits the results count. (When the limit is reached, the search stops).
+  playbookInputQuery:
+outputs:
+- contextPath: FireEyeHX.HuntingResults
+  description: Email message objects and fields that were retrieved from FireEye HX.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Timestamp - Event
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_File_Indicators_Hunting.yml:1961]: Should the contextPath have spaces?  Please check all contextPaths.

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Indicators_Hunting.yml [hunk 16]
```diff
@@ -0,0 +1,570 @@
...
+- key: limit
+  value: {}
+  required: false
+  description: Limits the results count. (When the limit is reached, the search stops).
+  playbookInputQuery:
+outputs:
+- contextPath: FireEyeHX.HuntingResults
+  description: Email message objects and fields that were retrieved from FireEye HX.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Timestamp - Event
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Indicators_Hunting.yml:514]: Should contextPath names have spaces? Please check all.

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Traffic_Indicators_Hunting.yml [hunk 17]
```diff
@@ -0,0 +1,1311 @@
...
+  required: false
+  description: |-
+    Whether a search is exhaustive.
+    Possible values are: yes, no. Default is yes.
+  playbookInputQuery:
+outputs:
+- contextPath: FireEyeHX.HuntingResults
+  description: Email message objects and fields that were retrieved from FireEye HX.
+  type: string
+- contextPath: FireEyeHX.HuntingResults.Timestamp - Event
```
Comment (ShirleyDenkberg) [inline Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Traffic_Indicators_Hunting.yml:1255]: Should contextPath names have spaces?  Please check all contextPaths.

**Comment by ShirleyDenkberg** [comment]
@altmannyarden Doc review completed.

File: Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Traffic_Indicators_Hunting.yml (hunk 17 above)
Comment (TalNos) [reply Packs/FireEyeHX/Playbooks/playbook-FireEye_HX_-_Traffic_Indicators_Hunting.yml:1255]: These are the outputs of the 'fireeye-hx-search (FireEyeHX v2)' command which contains spaces. 

Screenshot 2023-01-15 at 10 38 32

File: Packs/FireEyeETP/ReleaseNotes/1_1_0.md [hunk 18]
```diff
@@ -0,0 +1,5 @@
+
+#### Playbooks
+##### New: FireEye ETP - Indicators Hunting
+Created a new playbook for threat hunting and detection of IOCs such as domains, IP addresses, sender and recipient email addresses. 
+<!--Note that multiple search values should be separated by commas only (without spaces or any special characters). (Available from Cortex XSOAR 6.5.0).-->
```
Comment (altmannyarden) [inline Packs/FireEyeETP/ReleaseNotes/1_1_0.md:5]: @ShirleyDenkberg This note format isn't viewable. Is it on purpose?

Comment (TalNos) [reply Packs/FireEyeETP/ReleaseNotes/1_1_0.md:5]: I excluded it. 
The reason I excluded this sentence is that it repeats itself in every playbook description, and I thought it irrelevant to the RN.

**Review by altmannyarden — APPROVED** [review]

**Pull request merged**